src/parser/statement.js

```diff
@@ -1009,12 +1019,36 @@ export default class StatementParser extends ExpressionParser {
       return;
     }
 
+    const isPrivate = this.eat(tt.hash);
+
+    if (isPrivate) {
+      this.expectOnePlugin(["classPrivateProperties", "classPrivateMethods"]);
```

Comment:
I think we need to do this before eating the hash.

Reply:
Done.